examples/provider/main.tf

```diff
@@ -15,4 +15,5 @@ data "vellum_document_index" "reference" {
 
 resource "vellum_document_index" "managed" {
   label = "Managed Index"
+  name  = "managed-index" 
```

Comment:
@noanflaherty thoughts on `name` being optional in both the API and here, just as we do in our UI?

Reply:
My dream world is one where name is pulled automatically from the resource name `resource "vellum_document_index" "managed"` -> `"managed"`.

I don't think it should be optional. If anything, label should be.